ddtrace/contrib/asgi/middleware.py

```diff
@@ -259,9 +247,6 @@ async def wrapped_blocked_send(message):
                 self.handle_exception_span(exc, span)
                 reraise(exc_type, exc_val, exc_tb)
             finally:
-                try:
-                    del scope["datadog"]["request_span"]
-                except KeyError:
-                    pass
-
+                if span in scope["datadog"]["request_spans"]:
```

Comment:
Could you explain why we need to change this as well?

Reply:
This change looks to me like a less-verbose way of doing the same `del` logic that was on lines 262-265.

Reply:
It's a bit subtle but this is actually a "bug fix". scope["datadog"]["request_span"] is never set, we actually set scope["datadog"]["request_span**s**"].  This line always raises a KeyError. The impact of this bug is pretty low. We'll just hold more spans in memory than we need to. I can move this fix to it's own PR.